File: JS/LeetCode热题HOT100/11_盛最多水的容器.js
var maxArea = function(height) {
    let left = 0, right = height.length - 1;
    let res = 0
    while(left < right){
        if(height[left] <= height[right]){
            res = Math.max(res, height[left] * (right-left))
            left++
        }
        else{
            res = Math.max(res, height[right] * (right-left))
            right--
        }
    }
    return res
};
console.log(maxArea([1,8,6,2,5,4,8,3,7])); // 49
console.log(maxArea([1,1])); // 1

File: Python/面试经典150题/双指针/11_盛最多水的容器.py
from typing import List
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        ans = min(height[left], height[right]) * (right - left)
        while left < right:
            if height[left] < height[right]:
                left += 1
                if height[left] > height[left - 1]:
                    ans = max(ans, min(height[left], height[right]) * (right - left))
            else:
                right -= 1
                if height[right] > height[right + 1]:
                    ans = max(ans, min(height[left], height[right]) * (right - left))
        return ans

height = [1,8,6,2,5,4,8,3,7]
print(Solution().maxArea(height))
